script.js

```diff
@@ -1,4 +1,283 @@
+};
+
+var shuffleCards = function (cardDeck) {
+  var currentIndex = 0;
+  while (currentIndex < cardDeck.length) {
+    var randomIndex = getRandomIndex(cardDeck.length);
+    var randomCard = cardDeck[randomIndex];
+    var currentCard = cardDeck[currentIndex];
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard;
+    currentIndex = currentIndex + 1;
+  }
+  return cardDeck;
+};
+
+// setting rank as per black jack rules
+var counter = 0;
+while (counter < deck.length) {
+  if (
+    deck[counter].name == "jack" ||
+    deck[counter].name == "queen" ||
+    deck[counter].name == "king"
+  ) {
+    deck[counter].rank = 10;
+  }
+
+  counter = counter + 1;
+}
+
+// shuffle deck
+var shuffledDeck = shuffleCards(deck);
+
+// game modes  input name , player hand  ,
+var gameMode = "inputName";
+var playerName = "";
+var yourBetAmount = 0;
+var playerScore = 0;
+var card1 = {};
+var card2 = {};
+var dealerCards = [];
+
+var printCard = function (card) {
+  cardPic = `<div id="blackcard">${card.name}<br><span class="emoji">${card.emoji}</span></div>`;
+
+  if (card.suit == "hearts" || card.suit == "diamonds") {
+    cardPic = `<div id="redcard">${card.name}<br><span class="emoji">${card.emoji}</span></div>`;
+  }
+
+  return cardPic;
+};
+
+// modes inputName , betAmount, deal2cards , decidehit/Stay , dealerMode 2card autohit
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  if (gameMode == "inputName") {
+    playerName = input;
+    gameMode = "betAmount";
+    return (
+      "🥂 Hi " +
+      playerName +
+      ", welcome to the 'No Limit Blackjack Table'" +
+      "<br>" +
+      "👉 Please type in your betting amount 💰💰💰 in the orange bar above !"
+    );
+  }
+  if (gameMode == "betAmount") {
+    yourBetAmount = input;
+    gameMode = "ace1/11";
+    return (
+      "🥂 Great " +
+      playerName +
+      ", today seems to be your lucky day, 💰 you have bet: " +
+      yourBetAmount +
+      "<br>" +
+      "🤞 You can decide the Rank of Ace to be either 1 or 11" +
+      "<br>" +
+      "👉 Type in your choice: 1 or 11 in the orange bar above."
+    );
+  }
+
+  if (gameMode == "ace1/11") {
+    gameMode = "dealCards";
+    var counter = 0;
+    while (counter < shuffledDeck.length) {
+      if (shuffledDeck[counter].name == "ace") {
+        shuffledDeck[counter].rank == input;
+      }
+      counter = counter + 1;
+    }
+    return (
+      "🃎 Ace will have Rank: " +
+      input +
+      " for this round" +
+      "<br>" +
+      "♣️♠️♦️❤️ So it's now the Blackjack time. 👍 Press the 'Submit' button now" +
+      "<br>" +
+      "🃏🃏 Dealer will shuffle the deck and pull out two cards for you."
+    );
+  }
+
+  if (gameMode == "dealCards") {
+    card1 = shuffledDeck.pop();
+    card2 = shuffledDeck.pop();
+    playerScore = card1.rank + card2.rank;
+    if (playerScore == 21) {
+      gameMode = "endGame/newGame";
+      decision1 =
```

Comment:
Since `decision1` isn't a global variable I suggest properly creating it, like `var decision1 = ...`. Not declaring your variables properly will cause a bunch of problems, such as improper scoping of your `decision1` variable. You can read the StackOverflow discussion in the page I am about to link, starting with [this comment here](https://stackoverflow.com/questions/2485423/is-using-var-to-declare-variables-optional/2485439#2485439).

Since the `decision1` variable is only used in the `if` block when `(gameMode == "dealCards")`, I suggest creating the variable in this `if` block. Then in each `playerScore` conditional nested `if` block such as `if (playerScore == 21) { ... }`, re-assign `decision1`.

Consider the following:
```
if (gameMode == "dealCards") {
  var decision1 = "";
  card1 = shuffledDeck.pop();
  card2 = shuffledDeck.pop();
  playerScore = card1.rank + card2.rank;
  if (playerScore == 21) {
    gameMode = "endGame/newGame";
    decision1 = "perfect Blackjack , congrats" + "<br>" + "select end/continue ?";
  }
  ...
  ...
  ...
}
```